### DISCUSSION TITLE
Some questions about titanium alloy reconstruction in Example 2

### DISCUSSION THREAD

**Discussion by Cici-555** [ORTools4MTEX/ORTools#10] [Q&A] [answered]
When I use MTEX, I have a few issues that I don't understand. While I was using MTEX, I looked at the second example of titanium alloy reconstruction, and I had three questions.
 1. I do not understand whether the variation generated when I choose peak fit is deviating from Burgers relationship or conforming to Burgers relationship. 
 2. If I choose define instead of peak fitt, I choose Burgers orientation relationship, but the final crystal orientation and crystal plane are opposite, for example, Burgers relationship :(110) //(0001), [1-11] // [11-20]. When I use define, it comes out (1-11) // (11-20), [110] // [0001].
 3. In example 2, the number of variants is determined. How to determine the specific crystal planes and orientation corresponding to the variants?     These three problems have bothered me for more than a month and haven't been solved yet, so I can't go to the next step. I hope I can get a hint. Thank you.

**Comment by frankNiessen** [comment, marked as answer]
Hi Cici-555,

let me try and answer your questions:

1.	The fitted orientation relationship is very close to the Burgers orientation relationship but not exactly the same. The angular deviations I mention in the [tutorial video](https://youtu.be/e6R0dApUc8Q?t=273) are the angular deviations of the planes and directions of the fitted OR from the rational OR. You can see that it is tiny and practically speaking the OR’s therefore behave identically for the rest of the reconstruction.

2.	I have investigated this a bit: If you use the MTEX command “orientation.Burgers(job.csParent,job.csChild)” it will return the right Burgers OR. Internally, the computation of the OR looks like this:

```
mori = orientation.map(Miller(1,1,0,csParent),Miller(0,0,0,1,csChild),...
        Miller(-1,1,-1,csParent),Miller(2,-1,-1,0,csChild));
```

When you compute the disorientation angle between your manually defined OR and the MTEX Burgers OR for both the angle and axis, you will find that they both return 0, meaning that they are equivalent:

```
angle(job.p2c,orientation.Burgers(job.csParent,job.csChild))
angle(job.p2c.axis,orientation.Burgers(job.csParent,job.csChild).axis)
```

This means that the OR's are two different variants of the same OR and you can definitely use either of the OR’s to do the reconstruction. I must however admit that this is confusing and I will think of how to improve it. Do you have any input @AzdiarGazder ?

3.	The function ORinfo returns a structure with results: info = ORinfo(job). Here info.variants stores the orientations of the variants, the misorientations to variant 1 and the angles and axes of the variants. Applying `char(info.variants.orientation(1))` will give you the closest by rational OR of the first variant, `char(info.variants.orientation(2)) `of the second and so forth. Also this is not clear yet and I think we will have to add this to the list of future improvement, thank you for pointing it out. Any comment @AzdiarGazder ?

I hope this helps for now, 

Best wishes
Frank

**Reply by Cici-555** [reply]
Thank you very much for your answer! I have another problem. If I follow the peak fitting as shown in the video, I find that the Angle and axis of the variant are different from the five angular axis pairs of titanium alloys in the literature. If I follow your procedure in the video, what is the relationship between the calculated variant angular pairs and the five typical variant angular pairs? Finally, how can I determine the specific orientation and plane of the variant?

**Reply by Cici-555** [reply]
I would like to ask if my idea is correct. Mtex reconstructed titanium alloy with deviation from Burgers relationship to fit, and the variants calculated were not the 12 variants of Burgers relationship, but those that did not. As a result, the calculated variant has different angular axis pairs from the typical variant types of the five titanium alloys. But if so, does it show that the variation of Burgers relationship is not observed?

**Reply by frankNiessen** [reply]
> Thank you very much for your answer! I have another problem. If I follow the peak fitting as shown in the video, I find that the Angle and axis of the variant are different from the five angular axis pairs of titanium alloys in the literature. If I follow your procedure in the video, what is the relationship between the calculated variant angular pairs and the five typical variant angular pairs? Finally, how can I determine the specific orientation and plane of the variant?

So as I said in the previous post, you can get information on the variants by running `info = ORinfo(job.p2c)`. This will output the variant misorientations as angle axis pairs in the command window. In the case of the peak-fitted OR this is:

```
    - 1: 0.00º / [-0.5871,0.5871,0.0000,0.0000]
    - 2: 10.88º / [-0.0036,0.0018,0.0018,0.6409]
    - 3: 60.75º / [-0.3886,0.6665,-0.2779,0.1004]
    - 4: 59.89º / [-0.6779,0.3390,0.3390,0.0034]
    - 5: 60.75º / [-0.6665,0.3886,0.2779,0.1004]
    - 6: 63.31º / [-0.6455,0.3227,0.3227,0.1959]
    - 7: 60.86º / [-0.3887,0.6665,-0.2778,0.1003]
    - 8: 60.00º / [-0.6779,0.3390,0.3389,0.0034]
    - 9: 60.80º / [-0.6665,0.3877,0.2788,0.1009]
    - 10: 63.37º / [-0.3235,0.6454,-0.3218,0.1963]
    - 11: 90.00º / [-0.6749,0.3931,0.2818,0.0005]
    - 12: 90.06º / [-0.6748,0.3935,0.2813,0.0000]
```

You can then run `ORinfo(orientation.Burgers(job.csParent,job.csChild))` which should give you the Burgers OR as defined in literature. The angle axis output is:

```
  - 1: 0.00º / [0.5871,-0.0000,-0.5871,-0.0000]
  - 2: 60.00º / [-0.6779,0.3390,0.3390,-0.0000]
  - 3: 60.83º / [-0.6667,0.3865,0.2801,0.1006]
  - 4: 10.53º / [0.0000,-0.0000,-0.0000,0.6409]
  - 5: 60.83º / [-0.3865,0.6667,-0.2801,0.1006]
  - 6: 63.26º / [-0.6463,0.3232,0.3232,0.1934]
  - 7: 63.26º / [-0.6463,0.3232,0.3232,0.1934]
  - 8: 90.00º / [-0.3914,0.6751,-0.2837,-0.0000]
  - 9: 60.83º / [-0.6667,0.3865,0.2801,0.1006]
  - 10: 90.00º / [-0.3914,0.6751,-0.2837,-0.0000]
  - 11: 60.83º / [-0.3865,0.6667,-0.2801,0.1006]
  - 12: 60.00º / [-0.6779,0.3390,0.3390,-0.0000]
```

So you will be able to see that the variants are essentially the same, but that the order has changed as the definition of variant 1 is different here. Maybe you could share the list that you compare with?

When running `info = ORinfo(job.p2c)` you also get the info structure. Here you can also get the axis `info.variants.axis` and angle `info.variants.angle` directly as variables. To get the planes and directions you can run : 
`round2Miller(info.variants.orientation(1))` for any variant you like, which gives 

``` 
plane parallel   direction parallel    fit
(101) || (0001)   [-1-11] || [-12-10]     0.183°
```

for the first variants. 

I hope this brings you closer to what you want.

**Reply by frankNiessen** [reply]
> I would like to ask if my idea is correct. Mtex reconstructed titanium alloy with deviation from Burgers relationship to fit, and the variants calculated were not the 12 variants of Burgers relationship, but those that did not. As a result, the calculated variant has different angular axis pairs from the typical variant types of the five titanium alloys. But if so, does it show that the variation of Burgers relationship is not observed?

So if you do the fit you will see that the optimized OR is VERY close to Burgers OR, so in your case you can also do `job.p2c = orientation.Burgers(job.csParent,job.csChild` and it will not make any noticable difference. The variants that are calculated are the same 12 variants that you get with the Burgers OR (there might be 0.1° difference because of the slightly different OR). The ordering might be different, as there is an arbitrary first variant. If you want to follow some convention from literature you will have to list all twelve variants and make sure that they are ordered to your liking.

**Reply by Cici-555** [reply]
Thank you very much for your answer. The problem that has puzzled me for more than a month has finally been solved! Thanks again! In addition, I analyzed the transformation of titanium alloy texture in example 4, but I found that my β and α were opposite. I would like to know where the problem occurred. I have also watched your video teaching carefully, but I still don't know where the error occurred, resulting in the opposite of β and α. I hope it can be resolved. I'm sorry to bother you again.

**Reply by frankNiessen** [reply]
> Thank you very much for your answer. The problem that has puzzled me for more than a month has finally been solved! Thanks again! In addition, I analyzed the transformation of titanium alloy texture in example 4, but I found that my β and α were opposite. I would like to know where the problem occurred. I have also watched your video teaching carefully, but I still don't know where the error occurred, resulting in the opposite of β and α. I hope it can be resolved. I'm sorry to bother you again.

Glad I could help! I have just run through Example 4 myself again and everything seems to work as intended, so I cannot see exactly where your issue lies. 

What we do in example 4 is:

- Reconstruct beta from alpha
- Calculate the transformation texture based on the beta texture and no variant selection
- Compare the computed alpha transformation texture with the experimentally observed alpha texture
- Repeat the calculation with variant selection and variants weighting

**Reply by Cici-555** [reply]
Thank you for your answer! Let me check again to see if I made a mistake!  Thank you.

**Reply by Cici-555** [reply]
Dear Frank Niessen

I run the example 4 again and find that β is still the child phase and α is the parent phase. I think it may be because my material is metastable β titanium alloy, which precipitates α in the aging process, not (α+β) titanium alloy. If so, how can the texture of metastable β titanium alloys be predicted

**Reply by frankNiessen** [reply]
Hi Cici-555,

are you running example 4 with our default dataset or with your own data?
If you run with your own data it is your responsibility to assign the parent and the child phases to the function [setParentReconstructor](https://github.com/ORTools4MTEX/ORTools/blob/develop/src/setParentGrainReconstructor.m).

Are you having (1) a beta+alpha microstructure or (2) 100% alpha phase?
In case (1) you might already know your beta structure by the retained beta phase. 
In case (2) you can use the parent grain reconstruction as showcased in [example 2](https://github.com/ORTools4MTEX/ORTools#example-2-3-mar-2021) to obtain the beta microstructure.

I hope this helps, if the problem remains we would need more information about the dataset that you are working with. 

Best wishes
Frank
